### ISSUE TITLE
ERROR: Failed to pair with PC upon entering the PIN

### ISSUE THREAD

**Issue by Sagehawke** [KyroFrCode/moonlight-chrome-tizen#22]
I have successfully installed OneLiberty's version of Moonlight with updated UI on my Samsung TV to be able to enter my PC IP. When using either Sunshine or Gamestream on PC and selecting to open Steam on the TV, it pops up with a pin to enter on PC. I enter the pin on PC and it says successful and opens steam. Going back to the TV, it loads for a few seconds, screen goes black for a bit, then says connection terminated.

**Comment by OneLiberty** [comment]
I'll suggest sharing your sunshine logs and tv model number, and possibly pc specs. 
I had this issue when I let sunshine advertise for hevc I believe (don't quite remember)

Also, it's possible that this is related to my fork, trying the "original" version might help.

**Comment by Sagehawke** [comment]
Update: previous info is what happens when attempting to use Gamestream.
When using Sunshine, it immediately says "Error: failed to pair with (PC)" on the TV upon entering pin on PC.

TV model number: QN65Q70CAF

**Comment by Sagehawke** [comment]
Update: previous info is what happens when attempting to use Gamestream.
When using Sunshine, it immediately says "Error: failed to pair with (PC)" on the TV upon entering pin on PC.

Both devices are on the same network and both connected via ethernet.

I was not able to enter the IP on any other version of Moonlight due to the keyboard disappearing.

TV model number: QN65Q70CAF

SUNSHINE LOGS:
(_REMOVED_)

**Comment by OneLiberty** [comment]
Be sure to disable Gamestream before using Sunshine (reboot PC to be safe) 
If you can't pair, sunshine's logs are useless. 
I've just tested my latest version on a tizen 7.0 emulator, and I'm able to pair to the PC (It can take a few try as it does fail randomly, I don't really know why) 

If you have another device (phone or other computer) you can try another moonlight client (moonlight-qt or iOS/android port) to ensure that the issue is related to the tizen version

**Comment by Sagehawke** [comment]
I was finally able to connect! I switched back to the other version of Moonlight and was able to input the IP by using a keyboard and it worked after a few tries. Now my problem is that I the screen stays black. I can hear audio and move around the menu though :/

I have tested moonlight on my phone and it works fine.
Gamestream is uninstalled.
I have disabled advertizing HEVC in Sunshine.
I have unplugged my 2nd monitor.

**UPDATE**

I got it to work but only after fresh NVIDIA driver install, PC restart, and 720p 30FPS

**Comment by OneLiberty** [comment]
You've tested quite a lot of things ! 

So just to be sure, you weren't able to pin to the device using my fork, so you've used Kyro version and it worked ? 

Did you try multiple times on my fork as suggested ? 

Now for the blackscreen unfortunately, I have no answer for you, I've been investigating for a while now, but I'm not able to replicate the issue anymore, and don't have another tv to try on.
It's weird that it worked with 720@30 but no other combo.

So far the only thing that could fix it was a fresh driver install. 

I'll keep looking

**Comment by Sagehawke** [comment]
I tried to pin to it 15+ times, tweaking settings and restarting both devices many times and unfortunately it never worked with your fork. :/

I now have it running reliably with Kyro's fork and solved the black screen issue by lowering the Bitrate to 28mbps (default is 80). Now able to run 4K 60FPS with that lowered bitrate. I compensated for the lower bitrate by setting the encoder to max quality settings and have low to moderate input lag. It doesn't look that bad (as long as there isn't too much movement on screen).

**Comment by Sagehawke** [comment]
@OneLiberty I am willing to try again with your fork if you have any more ideas or updates.

**Comment by OneLiberty** [comment]
I'm glad you've managed to get it working ! 

That's weird for the pairing issue. If you're willing to try again, can you tell me if the pairing code changing when the dialog opens ? 

I've opened a testing branch where I try thing, it has the whole moonlight-common-c updated to match the one in moonlight chrome. 
I think, any issue related to my fork should be opened on it. This will be easier (at least for me) to track issues 

Also, wouldn't 1440p at a higher bitrate look better, isn't it fuzzy in motion with those settings ? 
Also are you familiar with Dev Tools ?

**Comment by anton123805** [comment]
Hi!
I installed your fork, and have the same issue, I tried to stream with sunshine nvenc or software encode, and geforce experience.
Previously i used KyroFr fork and it worked, but i have blackscreen while streaming with sunshine software encode.Because of this I tried your fork.
I have no issue with sunshine(no matter software encode or nvenc) or Geforce experience, it works perfectly fine, while i streaming to my Iphone with moonlight.

How can i try to install your testing branch?

Also i want to report about little bug, while entering IP numbers with TV remote and switching a little windows with numbers by pressing left and right buttons, you increase the number for +1 with clicking right button and -1 with clicking left.

**Comment by OneLiberty** [comment]
@anton123805 Multiple issues are discussed here, can you be more specific ? 

- Pin pairing ? 
- Blackscreen ? 

If you're talking about the blackscreen, did you try the possible fixes mentioned here ? (Bitrate ? Display drivers ?)
Also, could you share the TV Model Number, I'm trying to make a [list](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/7) about this to collect infos. 

For the testing branch you would have to use the "old" dockerfile method to install it. (Clone repo, docker build...) 

About the +1 -1 when entering the IP unfortunately I don't know why it's doing this with the remote but not with a gamepad. I've been unsuccessfully trying things for a week...

**Comment by anton123805** [comment]
@OneLiberty i have UE43CU8500UXRU.

- I have pin pairing problem on your fork, i tried to pair with sunshine and geforce experince.
Entering pin on sunshine/gamestream successfully, but moonlight failing to pair, tried many attempts.

- I have blackscreen in KuroFr fork if i stream with Gamestream or Sunshine software encoding. tried to fully reinstall display drivers and older versions, change encode settings, tried all resolutions from 720p(or 480p don't remember) to 4k,30 and 60 fps, hdr/non-hdr, from smallest bitrate to biggest, disabling and enabling AV1 and HEVC, changing quantization parameter and FEC Percentage.

Moonlight on my Iphone don't have any of this problems.
Thx for quick respond!

**Comment by OneLiberty** [comment]
@anton123805 
- For the pin pairing, could you try to unpair all clients in sunshine settings ? 
- Blackscreen issues are still weird in my opinion and i don't really know where to look, they are to many variables. 
Is your PC screen connected via DP or HDMI ? Might not be the issue as you said IOS didn't have any issue, but might be good to know.

**Comment by anton123805** [comment]
@OneLiberty 

- tried it
- DP

**Comment by anton123805** [comment]
@OneLiberty 

And one more problem. I have only one attempt to pair in moonlight, after failed attempt, i click connect and nothing happening until I restart moonlight

**Comment by OneLiberty** [comment]
@anton123805 
- Make sure to have the screen on when trying to stream then.
- You might have stacked some pairing request (maybe clicking multiple times or something...), could you try one more thing ? Mash the send button (in the "PIN pairing" page) in sunshine (for a minute or two) and try pairing again. If that doesn't work, i'm sorry but i have no ideas left. 

I'll try to make a guide on how to debug the app over on my fork so people can share their logs.

**Comment by anton123805** [comment]
@OneLiberty 

- screen is on and i don't have any problem when i stream with NVENC decode, but i don't have enough perfomance, I have more powerfull cpu than videocard, that's why i try to stream thru software. My suggestion, maybe something is off with decode algorithms(maybe client try to decode by one algorithm but stream is on another(just a shot in the dark)).

- I need to try mash send button after failed attempt? And i need to enter pin code or just mash button?

**Comment by OneLiberty** [comment]
> screen is on and i don't have any problem when i stream with NVENC decode, but i don't have enough perfomance, I have more powerfull cpu than videocard, that's why i try to stream thru software. My suggestion, maybe something is off with decode algorithms(maybe client try to decode by one algorithm but stream is on another(just a shot in the dark)).

Honnestly, no clue about that, i did try software encoder just now, and it worked fine on my end. (although it did spike my CPU at 40% usage, i'm not sure it's better than specific hardware encoder, but this ain't the subject)

> I need to try mash send button after failed attempt? And i need to enter pin code or just mash button?

After your first failed attempt, just mash the button on sunshine, i had this weird issue where it told "success" without anything entered, and after a while i was able to send a pairing request again.

**Comment by anton123805** [comment]
@OneLiberty 

Lol, it worked. I have pair now and stream on software decode is working.
But now i have another problems :/

- Tv is not switching to HDR mode and logs on sunshine said:
    Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
    Bits Per Color     : 10
    Red Primary        : [0.648438,0.339844]
    Green Primary      : [0.306641,0.636719]
    Blue Primary       : [0.149414,0.0498047]
    White Point        : [0.313477,0.329102]
    Min Luminance      : 0.0491 nits
    Max Luminance      : 408.76 nits
    Max Full Luminance : 322.118 nits
    [2023:12:16:14:36:24]: Info: SDR color coding [Rec. 601]
    [2023:12:16:14:36:24]: Info: Color depth: 8-bit
    [2023:12:16:14:36:24]: Info: Color range: [JPEG]
 No matter stream is on NVENC or Software. In KuroFr HDR worked(but only on NVENC(in Software i had blackscreen).
 
- If i stream thru Software and changing resolution on moonlight higher than 1080p i got stucked in "starting video steam establishment", i try to change bitrate from lowest to highest on client, changing setting of encode on server it had no effect. And after that i stuck on this stage even if i change resolution back to 1080p or lower, until i reboot client.
Also i try to change resolution to 1080p on server and connect with 4k on client and have stuck.

**Comment by OneLiberty** [comment]
@anton123805 

- How can you tell HDR is working ? I have no clue if it's working or not. I tried a few times, didn't see any differences between HDR and "non-HDR", so if you know where to look this might be helpful. 
Looking at the repo both Kyro and mine has  `//m_StreamConfig.enableHdr = true;` commented, which means HDR is not enabled, so expect if you've manually modified the line, if you've seen a difference it might have been placebo...

- Not sure what you're asking for here, the streamed resolution always match the host resolution. It means if you select 4K on client, but your desktop is 1080p, it will only be 1080 i believe, although someone might have to confirm that.

**Comment by anton123805** [comment]
@OneLiberty 

- idk maybe i used previous version(i used Dockerfile repo).
In TV menu of switching picture profiles was enabled game (hdr) profile, now just game profile.
In sunshine logs was something like:
[2023:12:16:14:36:24]: Info: HDR color coding 
[2023:12:16:14:36:24]: Info: Color depth: 10-bit
And when i enabling HDR with Kuro fork i have normal picture, now when i enabling HDR i have overexposed picture.
Can you help me, where is location of this line "//m_StreamConfig.enableHdr = true;" i change it for myself and how install it after that?

- I change HEVC and AV1 support to the last options and now i don't have "starting video steam establishment" problem.

> Not sure what you're asking for here, the streamed resolution always match the host resolution. It means if you select 4K on client, but your desktop is 1080p, it will only be 1080 i believe, although someone might have to confirm that.

maybe you didn't understand me, I streaming in 4k on desktop but client won't connect if i change resolution on moonlight client higher than 1080p(I have already fixed it as you can read at the top)

**Comment by OneLiberty** [comment]
@anton123805 
I've made a toggle to switch HDR on or off, based on my test it works, but for some reasons, the app run badly with this toggle (both on or off) could you try it ? 
Unfortunatly you'll have to build it yourself. 

[moonlight-chrome-tizen.zip](https://github.com/KyroFrCode/moonlight-chrome-tizen/files/13701702/moonlight-chrome-tizen.zip)

**Comment by anton123805** [comment]
@OneLiberty just tested about an hour. HDR is working now, don't saw any issues on my side.

**Comment by OneLiberty** [comment]
> @OneLiberty just tested about an hour. HDR is working now, don't saw any issues on my side.

really ? Stream is going wild after a minute or two on my side, did you try to play something ?

**Comment by anton123805** [comment]
@OneLiberty I played RDR 2 on 4k 60fps 80mbps HDR on NVENC encode.
Only one small issue i saw is a glitchy couple of frames which rarely happen(maybe somewhere once 10-15 minutes) and it doesn't bother me. It's not new trouble, it was when i used KuroFr fork too. Otherwise, everything is fine.

**Comment by anton123805** [comment]
@OneLiberty 

> Stream is going wild

What do you mean?

**Comment by OneLiberty** [comment]
@anton123805 

Well, it works fine the first 30seconds then it freezes every second or two, audio glitch out or just don't work I guess i just found another issue that is tv specific :/ tested on a S95B

If someone else confirm that it works ok, i'll push the update over on my fork despite my problem

**Comment by anton123805** [comment]
@OneLiberty Even without the load on the gpu?

**Comment by OneLiberty** [comment]
@anton123805 nvm, it's probably 'cause i had 2 different build of moonlight on my TV. 
Tested on ROR2 and BG3 via steam big picture, working fine now. 
We should probably stop commenting on this issue as it's not really related to pin pairing. 

For those who are looking for a workaround for failed paring on my fork https://github.com/KyroFrCode/moonlight-chrome-tizen/issues/22#issuecomment-1858787992 and https://github.com/KyroFrCode/moonlight-chrome-tizen/issues/22#issuecomment-1858792391 seemed to work.
